numba/runtime/nrt.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import print_function, absolute_import, division
+
+from . import _nrt_python as _nrt
+from . import atomicops
+
+
+class Runtime(object):
+    def __init__(self):
+        # Compile atomic operations
+        compiled = atomicops.compile_atomic_ops()
+        self._mcjit, self._ptr_inc, self._ptr_dec, self._ptr_cas = compiled
+        # Install atomic ops to NRT
+        _nrt.memsys_set_atomic_inc_dec(self._ptr_inc, self._ptr_dec)
+        _nrt.memsys_set_atomic_cas(self._ptr_cas)
+
+    def __del__(self):
+        _nrt.memsys_shutdown()
```

Comment:
Since the "memsys" is a singleton, the Runtime object can only be used once, right? If so, I think it should be instantiated as a global in this module. Otherwise, it's too easy to make a mistake.